srml/executive/src/lib.rs

```diff
@@ -96,18 +126,35 @@ impl<
 		// any initial checks
 		Self::initial_checks(&block);
 
-		// execute transactions
+		// execute extrinsics
 		let (header, extrinsics) = block.deconstruct();
-		extrinsics.into_iter().for_each(Self::apply_extrinsic_no_note);
-
-		// post-transactional book-keeping.
-		<system::Module<System>>::note_finished_extrinsics();
-		<AllModules as OnFinalise<System::BlockNumber>>::on_finalise(*header.number());
+		Self::execute_extrinsics_with_bookeeping(extrinsics, *header.number());
 
 		// any final checks
 		Self::final_checks(&header);
 	}
 
+	/// Execute all extrinsics like when executing a `block`, but with dropping intial and final checks.
+	pub fn execute_extrinsics_without_checks(block_number: NumberFor<Block>, extrinsics: Vec<Block::Extrinsic>) {
+		// Make the api happy, but maybe we should not set them at all.
+		let parent_hash = <Block::Header as Header>::Hashing::hash(b"parent_hash");
+		let extrinsics_root = <Block::Header as Header>::Hashing::hash(b"extrinsics_root");
+
+		Self::initialise_block_impl(&block_number, &parent_hash, &extrinsics_root);
+
+		// execute extrinsics
+		Self::execute_extrinsics_with_bookeeping(extrinsics, block_number);
+	}
+
+	/// Execute given extrinsics and take care of post-extrinsics book-keeping
```

Comment:
Should we mention that `initialisation` should be invoke manually prior to calling this function?

Reply:
As this is an internal function that reduces code duplication, I would omit this.